src/Facility.Definition/ServiceInfo.cs

```diff
@@ -30,10 +31,19 @@ public ServiceInfo(string name, IEnumerable<ServiceMemberInfo> members, IEnumera
 			foreach (var fieldGroup in GetDescendants().OfType<ServiceFieldInfo>().GroupBy(x => x.TypeName))
 			{
 				var type = ServiceTypeInfo.TryParse(fieldGroup.Key, FindMember);
-				if (type != null)
-					m_typesByName.Add(fieldGroup.Key, type);
-				else
+				if (type == null)
+				{
 					AddValidationErrors(fieldGroup.Select(x => new ServiceDefinitionError($"Unknown field type '{x.TypeName}'.", x.GetPart(ServicePartKind.TypeName)?.Position)));
+					return;
```

Comment:
Why do we skip the rest of the field groups?

Reply:
Whoops. Should be a continue.

Reply:
Without some compelling reason, I'd rather avoid `continue` and just use an `else` block.